### PR TITLE
Fixing FLOPS merge by checking if torch is available

### PR DESCRIPTION
**Pull request by LysandreJik**

### CODE REVIEW

File: src/transformers/trainer_utils.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
-import torch
 
 from .file_utils import is_tf_available, is_torch_available
 from .tokenization_utils_base import ExplicitEnum
 
 
+if is_torch_available():
+    import torch
+
+
```
Comment (LysandreJik) [inline src/transformers/trainer_utils.py:13]: @sgugger this isn't the *cleanest* as it isn't on parity with TensorFlow. I think this is acceptable until this file is refactored into two different files though?

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -1315,8 +1315,6 @@ def prediction_loop(
                 label_ids = xm.mesh_reduce("eval_label_ids", label_ids, torch.cat)
             if eval_losses is not None:
                 eval_losses = xm.mesh_reduce("eval_losses", torch.tensor(eval_losses), torch.cat).tolist()
-            if samples_count is not None:
```
Comment (sgugger) [inline src/transformers/trainer.py:1318]: Is this intentional?

File: src/transformers/trainer_utils.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/trainer_utils.py:13]: Yes, perfectly acceptable for now.

**Review by sgugger — APPROVED** [review]
Thanks for the quick fix!

File: src/transformers/trainer.py (hunk 2 above)
Comment (LysandreJik) [reply src/transformers/trainer.py:1318]: Yes, removed artefact from FLOPS PR (this runs on TPU and wasn't caught by the test).

**Pull request merged**